src/components/user-view/raw-text-document/huge-rte/index.tsx

```diff
@@ -0,0 +1,212 @@
+  const [loading, setLoading] = useState<boolean>(!!docId);
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [tempTitle, setTempTitle] = useState('');
+  useWithRawTextDocVersions(currentActivityId, docData);
+  const { updateDocTitleLocally } = useWithUsersDocs();
+
+  const editor = useRef<Editor>(null);
+
+  const editorConfig = useMemo(
+    () => ({
+      plugins: ['markdown', 'lists'],
+      toolbar: [
+        { name: 'history', items: ['undo', 'redo'] },
+        { name: 'styles', items: ['styles'] },
+        { name: 'formatting', items: ['bold', 'italic'] },
+        {
+          name: 'alignment',
+          items: ['alignleft', 'aligncenter', 'alignright', 'alignjustify'],
+        },
+        { name: 'indentation', items: ['outdent', 'indent'] },
+        { name: 'lists', items: ['unordered', 'ordered'] },
+      ],
+    }),
+    []
+  );
+
+  // Debounced update function
+  const debouncedUpdate = useMemo(
+    () =>
+      debounce((markdownText: string) => {
+        const rawText = editor.current?.editor?.getContent({ format: 'text' });
+        if (docData) {
+          setDocData({
+            ...docData,
+            plainText: rawText || markdownText,
+            markdownText: markdownText,
+            lastChangedId: docData.lastChangedId || '',
+          });
+        }
+      }, 500),
+    [docData]
+  );
+
+  useEffect(() => {
+    if (docId) {
+      setLoading(true);
+      getDocData(docId, DocService.RAW_TEXT)
+        .then((docData) => {
+          setDocData(docData);
+          setInitialDocData(docData);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }, [docId]);
+
+  // Cleanup debounce on unmount
+  useEffect(() => {
+    return () => {
+      debouncedUpdate.cancel();
+    };
+  }, [debouncedUpdate]);
+
+  const handleEditorChange = useCallback(
+    (value: string) => {
+      debouncedUpdate(value);
+    },
+    [debouncedUpdate]
+  );
+
+  const handleTitleChange = async (newTitle: string) => {
+    if (docId && docData) {
+      try {
+        setDocData({ ...docData, title: newTitle });
+        setIsEditingTitle(false);
+        updateDocTitleLocally(docId, newTitle);
+      } catch (error) {
+        console.error('Error updating document title:', error);
+      }
+    }
+  };
+
+  const handleCancelEdit = () => {
+    if (docData) {
+      setTempTitle(docData.title);
+      setIsEditingTitle(false);
+    }
+  };
+
+  const handleSaveEdit = () => {
+    handleTitleChange(tempTitle);
+  };
+
+  const MemoizedEditor = useMemo(
+    () => (
+      <Editor
+        initialValue={initialDocData?.markdownText}
+        ref={editor}
+        onChange={(value) => {
+          console.log(value.target);
+        }}
+        plugins={editorConfig.plugins}
+        toolbar={editorConfig.toolbar}
+        onEditorChange={handleEditorChange}
+      />
+    ),
+    [initialDocData]
```

Comment:
you should pass `editor`, `editorConfig`, and `handleEditorChange` in dependency array too

Reply:
editorConfig never changes so no need to put in dependency array; i updated and put the other 2 in